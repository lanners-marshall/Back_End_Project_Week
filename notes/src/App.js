import React, { Component } from 'react';
import { Route, withRouter, Switch } from 'react-router-dom';
import { BGColor, City } from './css';
import SideBar from './components/SideBar/SideBar';
import axios from 'axios'

import NoteList from './components/NoteInfo/NoteList';
import CreateNote from './components/NoteInfo/CreateNote'
import LinkedNote from './components/NoteInfo/LinkedNote';
import Weather from './components/Weather/Weather'
import Signin from './components/Auth/Signin'
import SignUp from './components/Auth/SignUp';

class App extends Component {
  constructor() {
    super();
    this.state = {
      topNews: [],
    };
  }


  componentDidMount() {
    axios
      .get("https://newsapi.org/v2/top-headlines?country=us&apiKey=fe1ec32653734ee8895f2371139ed9da")
      .then(response => {

        let topTen = [];

        for (let i = 0; i < 3; i++){
          topTen.push(response.data.articles[i])
        }

        //console.log(topTen)

        this.setState({topNews: topTen})
      })
      .catch(err => {
        console.log(err)
      })
  }

  handleData = data => this.setState({ notesData: data });

  render() {
    return (
      <BGColor>
        <Route
          path="/notes"
          render={props => (
          <div>
            <SideBar
              {...props}
              news={this.state.topNews}
            />
          </div>
          )}
        />

        <Route
          exact
          path="/notes"
          render={props => (
          <div>
            <City>{this.state.city}</City>
            <Weather
              {...props}
            />
          </div>
          )}
        />

        <Route
          exact
          path="/notes"
          render={props => (
          <div>
            <NoteList
              {...props}
            />
          </div>
          )}
        />

      <Switch>
        <Route
          path="/notes/create"
          render={props => (
          <div>
            <CreateNote
              {...props}
              handleData={this.handleData}
            />
          </div>
          )}
        />

        <Route
          exact
          path="/notes/:id"
          render={props => (
          <div>
            <LinkedNote
              {...props}
              handleData={this.handleData}
            />
          </div>
          )}
        />
      </Switch>

        <Route exact path='/signin' component={Signin} />
        <Route exact path='/signup' component={SignUp} />
      </BGColor>
    );
  }
}

export default withRouter(App);
