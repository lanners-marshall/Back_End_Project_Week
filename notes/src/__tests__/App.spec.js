import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter, Route } from 'react-router-dom';

import App from '../App';
import renderer from 'react-test-renderer';

describe('<App />', () => {
  it('should match snapshot', () => {
	  const wrapper = mount(
	  	<MemoryRouter initialEntries={['/notes']}>
	    <App />
	    </MemoryRouter>
  	)
  	const tree = renderer.create(wrapper).toJSON()
  	expect(tree).toMatchSnapshot()
	})
});