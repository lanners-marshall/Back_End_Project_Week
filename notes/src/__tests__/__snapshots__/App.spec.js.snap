// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="sc-bdVaJa ekoYDo"
>
  <div>
    <div
      className="sc-htpNat hfVMnL"
    >
      <h1>
        Lambda
        <br />
         Notes
      </h1>
      <a
        href="/notes"
        onClick={[Function]}
      >
        <div
          className="sc-bxivhb cMpcrm"
        >
          <p>
            View Your Notes
          </p>
        </div>
      </a>
      <a
        href="/notes/create"
        onClick={[Function]}
      >
        <div
          className="sc-bxivhb cMpcrm"
        >
          <p>
            + Create New Note
          </p>
        </div>
      </a>
      <div
        className="sc-bxivhb cMpcrm"
        onClick={[Function]}
      >
        <p>
          Log Out
        </p>
      </div>
      <h2
        className="sc-EHOje bNtLXl"
      >
        Current News
      </h2>
      <a
        href="http://newsapi.org"
      >
        <p
          className="sc-gzVnrw hjqJpX"
        >
          - newsapi.org
        </p>
      </a>
    </div>
  </div>
  <div>
    <div
      className="sc-bwzfXH lmzXpv"
    />
    <div>
      <div
        className="sc-jAaTju jasGxi"
      >
        
      </div>
      <div
        className="sc-jWBwVP dYzgoT"
      />
      <div
        className="sc-cMljjf kTeYqF"
      >
        <form>
          <input
            name="inputCity"
            onChange={[Function]}
            placeholder="City"
          />
          <input
            name="country"
            onChange={[Function]}
            placeholder="Country Code, us, uk ect"
            value=""
          />
        </form>
        <button
          onClick={[Function]}
        >
          check your local weather
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      className="sc-eHgmQL Vjfqi"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          className="sc-cvbbAY ksyAOn"
          onChange={[Function]}
          placeholder="Search by Author"
        />
      </form>
      <h2>
        Your Notes:
      </h2>
      <div
        className="sc-hSdWYo eKFpHH"
      />
    </div>
  </div>
</div>
`;
